lms/djangoapps/courseware/views/index.py

```diff
@@ -435,7 +438,13 @@ def _create_courseware_context(self, request):
                 table_of_contents['previous_of_active_section'],
                 table_of_contents['next_of_active_section'],
             )
-            courseware_context['fragment'] = self.section.render(STUDENT_VIEW, section_context)
+
+            # block content
+            if request.user.is_authenticated:
+                courseware_context['fragment'] = self.section.render(STUDENT_VIEW, section_context)
+            elif self.enable_anonymous_courseware_access and self.course.course_visibility == 'public':
```

Comment:
We don't need all these checks again here. Following the pattern of how the other parameters are being initialized, set `self.view = STUDENT_VIEW if some check else ANONYMOUS_VIEW` in `get()`.